### ISSUE TITLE
Rule Proposal: Disallow properties starting with `:v-`.

### ISSUE THREAD

**Issue by mikegreiling** [vuejs/eslint-plugin-vue#935]
**Please describe what the rule should do:**

I have seen an occasion where a `v-if` was mistakenly typed as `:v-if` and the error was not immediately obvious.  Since it is unlikely that someone will ever legitimately want to name a property "v-*" we could make a rule to catch these simple typos.

**What category should the rule belong to?**

- [ ] Enforces code style
- [x] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
```vue
<some-component :v-if="someCondition" />
```

**Additional context**

Related issue (on GitLab): https://gitlab.com/gitlab-org/gitlab-ce/merge_requests/31222#note_197281204
